File: .github/scripts/dependency/Plan.py
"""A complete dependency update plan."""

from __future__ import annotations

from dataclasses import dataclass

from dependency.Change import Change


@dataclass(frozen=True, slots=True)
class Plan:
    """The complete in-memory Dockerfile update."""

    content: str
    changes: tuple[Change, ...]

    @property
    def changed(self) -> bool:
        return any(change.changed for change in self.changes)

File: .github/scripts/src/Dependency/Plan.php
<?php

declare(strict_types=1);

namespace DockerBase\Dependency;

final readonly class Plan
{
    /**
     * @param list<Change> $changes
     */
    public function __construct(
        public string $content,
        public array $changes,
    ) {
    }

    public function changed(): bool
    {
        foreach ($this->changes as $change) {
            if ($change->changed()) {
                return true;
            }
        }

        return false;
    }
}
